benchflow-experiment-manager/application/src/main/java/cloud/benchflow/experimentmanager/tasks/running/CheckTrialResultTask.java

```diff
@@ -67,22 +71,23 @@ public TrialResult call() throws Exception {
 
       }
 
-      if (trialResult == TrialResult.FAILURE && retries >= maxRetries) {
-        trialResult = TrialResult.EXECUTION_FAILURE;
+      if (failureStatus == null) {
```

Comment:
the controller decides what to do next - here we just provide the values in order to make the decision